### PR TITLE
Add exponential backoff retry

### PR DESCRIPTION
**Pull request by CoveMB**
Add exponential backoff retry config.

retries: number of retry
retryCondition: function to evaluate if should retry 
(default to retrying on cloudflare errors, auth error not counted as retry)
retryDelay: to evaluate the retry delay default to exponential backoff

### CODE REVIEW

File: packages/base/src/api/client.ts [hunk 1]
```diff
@@ -57,31 +73,67 @@ export abstract class BaseApiClient {
...
+
+export const exponentialDelay = (
+  retryNumber = 0,
+  _error: AxiosError | undefined = undefined,
+  delayFactor = 100,
+): number => {
+  const delay = 2 ** retryNumber * delayFactor;
+  const randomSum = delay * 0.2 * Math.random(); // 0-20% of the delay
+  return delay + randomSum;
+};
```
Comment (MCarlomagno) [inline packages/base/src/api/client.ts:139]: I think the idea was to use [axios retry](https://www.npmjs.com/package/axios-retry) to abstract all this logic, is there a reason why you did the custom implementation?

Comment (CoveMB) [reply packages/base/src/api/client.ts:139]: Yea, axios retry does not allow modifying the request (for cloudflare upgrade headers for example) also we qwould need to refactor much of how the api is managed if you follow the axios instance it get's recreated on refresh

**Review by tirumerla — APPROVED** [review]
LGTM

File: package.json [hunk 2]
```diff
@@ -65,4 +65,4 @@
   "resolutions": {
     "@ethersproject/abstract-provider": "^5.6.1"
   }
-}
+}
```
Comment (tirumerla) [inline package.json:68]: Nit: Eol

**Pull request merged**